src/services/item/plugins/importExport/service.test.ts

```diff
@@ -59,4 +60,192 @@ describe('ZIP routes tests', () => {
+              path: 'path',
+              size: 111,
+              content: 'content',
+            },
+          },
+        },
+      });
+      const importExportService = new ImportExportService(
+        app.db,
+        {
+          getUrl: jest.fn(),
+        } as unknown as FileItemService,
+        resolveDependency(ItemService),
+        {} as unknown as H5PService,
+        resolveDependency(BaseLogger),
+      );
+      const repositories = buildRepositories();
+      const res = await importExportService.fetchItemData(actor, repositories, item);
+
+      expect(res.name).toEqual(item.name + '.png');
+      expect(res.stream).toBeDefined();
+      expect(res.mimetype).toEqual('image/png');
+    });
+    it('fetch app data', async () => {
+      ({ app, actor } = await build());
+      const { item } = await testUtils.saveItemAndMembership({
+        member: actor,
+        item: {
+          type: ItemType.APP,
+        },
+      });
+      const importExportService = new ImportExportService(
+        app.db,
+        {} as unknown as FileItemService,
+        resolveDependency(ItemService),
+        {} as unknown as H5PService,
+        resolveDependency(BaseLogger),
+      );
+      const repositories = buildRepositories();
+      const res = await importExportService.fetchItemData(actor, repositories, item);
+
+      expect(res.name).toEqual(item.name + '.app');
+      expect(res.buffer).toBeDefined();
+      expect(res.mimetype).toEqual('text/plain');
+    });
+    it('fetch link data', async () => {
+      ({ app, actor } = await build());
+      const { item } = await testUtils.saveItemAndMembership({
+        member: actor,
+        item: {
+          type: ItemType.LINK,
+        },
+      });
+      const importExportService = new ImportExportService(
+        app.db,
+        {} as unknown as FileItemService,
+        resolveDependency(ItemService),
+        {} as unknown as H5PService,
+        resolveDependency(BaseLogger),
+      );
+      const repositories = buildRepositories();
+      const res = await importExportService.fetchItemData(actor, repositories, item);
+
+      expect(res.name).toEqual(item.name + '.url');
+      expect(res.buffer).toBeDefined();
+      expect(res.mimetype).toEqual('text/plain');
+    });
+    it('fetch document data', async () => {
+      ({ app, actor } = await build());
+      const { item } = await testUtils.saveItemAndMembership({
+        member: actor,
+        item: {
+          type: ItemType.DOCUMENT,
+        },
+      });
+      const importExportService = new ImportExportService(
+        app.db,
+        {} as unknown as FileItemService,
+        resolveDependency(ItemService),
+        {} as unknown as H5PService,
+        resolveDependency(BaseLogger),
+      );
+      const repositories = buildRepositories();
+      const res = await importExportService.fetchItemData(actor, repositories, item);
+
+      expect(res.name).toEqual(item.name + '.graasp');
```

Comment:
Interesting, so when the isRaw setting is true then we use the `.html` format, but when it is rich text, we use the `.graasp` format, even when it is also valid HTML. I am not sure how we could make it so that both documents can be easily opened on the computer and we are still able to re-import them correctly.

Reply:
Actually I could use html in any case. `.graasp` is legacy. Let's consider with this issue: https://github.com/graasp/graasp/issues/433